src/common/types/column/column_data_collection_segment.cpp

```diff
@@ -14,10 +14,31 @@ idx_t ColumnDataCollectionSegment::GetDataSize(idx_t type_size) {
 	return AlignValue(type_size * STANDARD_VECTOR_SIZE);
 }
 
-validity_t *ColumnDataCollectionSegment::GetValidityPointer(data_ptr_t base_ptr, idx_t type_size) {
+validity_t *ColumnDataCollectionSegment::GetValidityPointerForWriting(data_ptr_t base_ptr, idx_t type_size) {
 	return reinterpret_cast<validity_t *>(base_ptr + GetDataSize(type_size));
 }
 
+validity_t *ColumnDataCollectionSegment::GetValidityPointer(data_ptr_t base_ptr, idx_t type_size, idx_t count) {
+	auto validity_mask = reinterpret_cast<validity_t *>(base_ptr + GetDataSize(type_size));
+
+	// Optimized check to see if all entries are valid
+	for (idx_t i = 0; i < (count / ValidityMask::BITS_PER_VALUE); i++) {
+		if (!ValidityMask::AllValid(validity_mask[i])) {
+			return validity_mask;
+		}
+	}
+	if ((count % ValidityMask::BITS_PER_VALUE) != 0) {
+		auto entry = validity_mask[(count / ValidityMask::BITS_PER_VALUE)];
```

Comment:
Maybe we could do this with a single bitwise operation?

Reply:
Hmm constructing the mask and then comparing against that in one go?

Reply:
Yeah I was wondering if some kind of shifted mask comparison would work: `entry & mask = mask`. I think mask could be constructed from all bits shifted by the ragged count?